cypress/e2e/claim-portal-non-web3.cy.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable sonarjs/no-duplicate-string */
+describe("Claims Portal Non-Web3", () => {
+  beforeEach(() => {
+    cy.clearAllCookies();
+    cy.clearAllLocalStorage();
+    cy.clearAllSessionStorage();
+
+    setupIntercepts();
+
+    cy.visit(`/${claimUrl}`);
+    cy.wait(2000);
+  });
+
+  describe("No window.ethereum", () => {
+    it("Should toast and hide buttons in a non-web3 env", () => {
+      cy.get("#invalidator").should("not.be.visible");
+      cy.get("#claim-loader").should("not.be.visible");
+      cy.get("#view-claim").should("not.be.visible");
+
+      cy.get("body", { timeout: 3000 }).should("contain.text", "Please use a web3 enabled browser to collect this reward.");
+    });
+  });
+
+  describe("Mobile: No window.ethereum", () => {
+    beforeEach(() => {
+      cy.viewport("iphone-6");
+      cy.reload();
+    });
+
+    it("Should toast and hide buttons in a non-web3 env", () => {
+      cy.get("#invalidator").should("not.be.visible");
+      cy.get("#claim-loader").should("not.be.visible");
+      cy.get("#view-claim").should("not.be.visible");
+
+      cy.get("body", { timeout: 3000 }).should("contain.text", "Please use a web3 enabled browser to collect this reward.");
+    });
+  });
+});
+
+function setupIntercepts() {
+  cy.intercept("POST", "*", (req) => {
+    // return a 404 for rpc optimization meaning no successful RPC
+    // to return our balanceOf and allowance calls
+    if (req.body.method === "eth_getBlockByNumber") {
+      req.reply({
+        statusCode: 404,
+        body: {
+          jsonrpc: "2.0",
+          error: {
+            code: -32601,
+            message: "Method not found",
+          },
+          id: 1,
+        },
+      });
+    }
+  });
+
+  cy.intercept("POST", "https://gnxgtwvmduxcwucovxqp.supabase.co/rest/v1/*", {
```

Comment:
Which db is this targeting?

Reply:
my own local I will change to the current DB, my bad